hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientMembershipListener.java

```diff
@@ -141,6 +143,18 @@ void listenMembershipEvents(Connection ownerConnection) throws Exception {
         waitInitialMemberListFetched();
     }
 
+    public void clearMemberListOnClusterRestart() {
+        Set<Member> prevMembers = Collections.emptySet();
+        if (!members.isEmpty()) {
```

Comment:
Can `members` be null here?

Reply:
I was thinking it can not since we are resetting cluster id  to null when we set `members` to null. But if an authentication fails after setting cluster id, it could be the case that second authentication hits NullPointerException here. 

I have changed solution to not use `null` but a static place holder.